added test for send_frame_options_header

pbearne opened the pull request:
Trac ticket: https://core.trac.wordpress.org/ticket/59851

pbearne committed:
added test for send_frame_options_header

swissspidy commented on tests/phpunit/tests/functions/sendFrameOptionsHeader.php:
```diff
@@ -0,0 +1,22 @@
+/**
+ * Tests for the send_frame_options_header function.
+ *
+ * @group functions.php
```
```suggestion
 * @group functions
```

swissspidy commented on tests/phpunit/tests/functions/sendFrameOptionsHeader.php:
```diff
@@ -0,0 +1,22 @@
+ *
+ * @covers ::send_frame_options_header
+ */
+class Tests_functions_sendFrameOptionsHeader extends WP_UnitTestCase{
```
```suggestion
class Tests_Functions_SendFrameOptionsHeader extends WP_UnitTestCase {
```

swissspidy commented on tests/phpunit/tests/functions/sendFrameOptionsHeader.php:
```diff
@@ -0,0 +1,22 @@
+	 *
+	 * @ticket 59851
+	 */
+	public function test_send_frame_options_header() {
+
+		$this->assertTrue( function_exists('send_frame_options_header') );
```
This is not really a useful test in itself.

To make this test useful, use `xdebug_get_headers()` and `@runInSeparateProcess` to check that the correct header was sent. See existing core tests for precedence

pbearne committed:
Update tests/phpunit/tests/functions/sendFrameOptionsHeader.php

Co-authored-by: Pascal Birchler <pascal.birchler@gmail.com>

pbearne committed:
Update tests/phpunit/tests/functions/sendFrameOptionsHeader.php

Co-authored-by: Pascal Birchler <pascal.birchler@gmail.com>

pbearne committed:
Adjust unit test for send_frame_options_header function

The unit test for send_frame_options_header function is updated to manage PHP Warnings and check response headers. setUp and tearDown methods are added to manage error levels, and the test_send_frame_options_header method now checks the 'X-Frame-Options' in the response headers. Also, this update necessitated the inclusion of xdebug as part of the test suite.

pbearne committed:
Adjust unit test for send_frame_options_header function

The unit test for send_frame_options_header function is updated to manage PHP Warnings and check response headers. setUp and tearDown methods are added to manage error levels, and the test_send_frame_options_header method now checks the 'X-Frame-Options' in the response headers. Also, this update necessitated the inclusion of xdebug as part of the test suite.

pbearne committed:
Remove unnecessary whitespace from PHPUnit test

Two superfluous line breaks were removed from the PHPUnit test for sendFrameOptionsHeader function to improve code readability. This refactoring does not alter the functionality or performance of the test suite.

pbearne committed:
Update assertStringContainsString statement in PHPUnit test

This patch adjusts the formatting of the assertStringContainsString statement in the PHPUnit test for the sendFrameOptionsHeader function. It simply removes an extra space without making substantial changes to the code or its functionality.

pbearne committed:
Update assertStringContainsString statement in PHPUnit test

This patch adjusts the formatting of the assertStringContainsString statement in the PHPUnit test for the sendFrameOptionsHeader function. It simply removes an extra space without making substantial changes to the code or its functionality.

pbearne committed:
debug

pbearne committed:
debug

pbearne committed:
Remove debug statement from test file

The debug statement (`var_dump( $headers );`) was removed from the sendFrameOptionsHeader PHPUnit test. This modification makes the test output cleaner and more focused on asserting the expected conditions.

pbearne committed:
Remove unneeded error reporting in sendFrameOptionsHeader test

The line setting the error level was removed from the sendFrameOptionsHeader PHPUnit test. This tweak simplifies the test setup and helps to reduce the side effects associated with modifying the global error reporting level.

pbearne committed:
Simplify sendFrameOptionsHeader test by removing error suppression

The set_up method in sendFrameOptionsHeader test was modified to get rid of the error suppression line. This change helps decrease the overall complexity of the test setup. Moreover, it minimizes potential negative impacts related to altering the global error reporting level.

swissspidy commented on tests/phpunit/tests/functions/sendFrameOptionsHeader.php:
```diff
@@ -0,0 +1,34 @@
+ */
+class Tests_Functions_SendFrameOptionsHeader extends WP_UnitTestCase {
+
+
```
```suggestion
```

swissspidy commented on tests/phpunit/tests/functions/sendFrameOptionsHeader.php:
```diff
@@ -0,0 +1,34 @@
+	 * @requires function xdebug_get_headers
+	 */
+	public function test_send_frame_options_header() {
+		$this->assertTrue( function_exists( 'send_frame_options_header' ) );
```
Can you elaborate on this assertion? When should this function not exist?

pbearne replied:
I will remove it 
lets over from the old version

pbearne committed:
Update tests/phpunit/tests/functions/sendFrameOptionsHeader.php

Co-authored-by: Pascal Birchler <pascal.birchler@gmail.com>

pbearne committed:
Update sendFrameOptionsHeader.php

remove not needed test
